timesketch/lib/analyzers/yetiindicators.py

```diff
@@ -39,48 +27,58 @@ def __init__(self, index_name, sketch_id):
         self.intel = {}
         self.yeti_api_root = current_app.config.get('YETI_API_ROOT')
         self.yeti_api_key = current_app.config.get('YETI_API_KEY')
-        self.yeti_indicator_labels = current_app.config.get(
-            'YETI_INDICATOR_LABELS', [])
 
-    def get_bad_domain_indicators(self, entity_id):
-        """Retrieves a list of indicators associated to a given entity.
+    def get_neighbors(self, entity_id):
+        """Retrieves a list of neighbors associated to a given entity.
 
         Args:
           entity_id (str): STIX ID of the entity to get associated inticators
-                from. (typically an Intrusion Set)
+                from. (typically an Intrusion Set or an Incident)
 
         Returns:
-          A list of JSON objects describing a Yeti Indicator.
+          A list of JSON objects describing a Yeti object.
         """
         results = requests.post(
             self.yeti_api_root + '/entities/{0:s}/neighbors/'.format(entity_id),
             headers={'X-Yeti-API': self.yeti_api_key},
         )
         if results.status_code != 200:
             return []
-        domain_indicators = []
+        neighbors = []
         for neighbor in results.json().get('vertices', {}).values():
-            if neighbor['type'] == 'x-regex' and \
-                set(self.yeti_indicator_labels) <= set(neighbor['labels']):
-                domain_indicators.append(neighbor)
-
-        return domain_indicators
+            neighbors.append(neighbor)
 
-    def get_intrusion_sets(self):
-        """Populates the intel attribute with data from Yeti.
+        return neighbors
 
-        Retrieved intel consists of Intrusion sets and associated Indicators.
-        """
+    def get_indicators(self, indicator_type):
+        """Populates the intel attribute with entities from Yeti."""
         results = requests.post(
-            self.yeti_api_root + '/entities/filter/',
-            json={'name': '', 'type': 'intrusion-set'},
+            self.yeti_api_root + '/indicators/filter/',
+            json={'name': '', 'type': indicator_type},
             headers={'X-Yeti-API': self.yeti_api_key},
         )
         if results.status_code != 200:
             return
         self.intel = {item['id']: item for item in results.json()}
-        for _id in self.intel:
-            self.intel[_id]['indicators'] = self.get_bad_domain_indicators(_id)
+        for item in results.json():
+            item['compiled_regexp'] = re.compile(item['pattern'])
+            self.intel[item['id']] = item
+
+    def mark_event(self, indicator, event, neighbors):
+        """Anotate an event with data from indicators and neighbors.
+
+        Tags with skull emoji, adds a comment to the event.
+        """
+        event.add_emojis([emojis.get_emoji('SKULL')])
+        event.add_tags([n['name'] for n in neighbors])
```

Comment:
Out of curiosity, what is the name of a neighbour? Just wanna make sure we don't end up with too long tag names which will clutter the UI.

Reply:
It's the human-readable name of whatever neighbor it is. As an example, the names of TTPs imported through MITRE ATT&CK are the names of the actual TTPs: https://attack.mitre.org/

So yeah, tag names could end up being long? More importantly (and I just though about this) they could also contain spaces, which might make searching for them a bit tricky. WDYT? We could slugify the names

Reply:
Yeah, sluggify would be great!
I rather do it now, before any saved views etc wil depend on the tag name